From 15f2ddb7175008d14b25b7ab2eb71b487a497327 Mon Sep 17 00:00:00 2001
From: Naushir Patuck <naush@raspberrypi.com>
Date: Tue, 19 May 2020 16:56:33 +0100
Subject: [PATCH] media: i2c: imx477: Return correct result on sensor
 id verification

The test should return -EIO if the register read id does not match
the expected sensor id.

Signed-off-by: Naushir Patuck <naush@raspberrypi.com>
---
 drivers/media/i2c/imx477.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/media/i2c/imx477.c
+++ b/drivers/media/i2c/imx477.c
@@ -1919,7 +1919,7 @@ static int imx477_identify_module(struct
 	if (val != IMX477_CHIP_ID) {
 		dev_err(&client->dev, "chip id mismatch: %x!=%x\n",
 			IMX477_CHIP_ID, val);
-		ret = -EINVAL;
+		return -EIO;
 	}
 
 	return 0;
